File: 3100. Water Bottles II.java
class Solution {
    public int maxBottlesDrunk(int numBottles, int numExchange) {
        int answer = 0;
        int empty = 0;
        while (numBottles > 0) {
            answer += numBottles;
            empty += numBottles;
            numBottles = 0;
            while (empty >= numExchange) {
                numBottles += 1;
                empty -= numExchange;
                numExchange += 1;
            }
        }
        return answer;
    }
}

File: 3100. Water Bottles II.go
func maxBottlesDrunk(numBottles int, numExchange int) int {
	answer := 0
	empty := 0
	for numBottles > 0 {
		answer += numBottles
		empty += numBottles
		numBottles = 0
		for empty >= numExchange {
			empty -= numExchange
			numBottles += 1
			numExchange += 1
		}
	}
	return answer
}